File: Tareas/T11SerieB.java
/*  NOMBRE : OSCAR IGNACIO GUAJARDO HERNANDEZ
MATRICULA : 1865021  HORA: V4-6;*/

import javax.swing.JOptionPane;

public class T11SerieB
{
	public static void main (String[] arg)
	{
		String mensaje="";
		int pos=0, i=1, j=2, posdes=0, k=1;
		String Ventana = JOptionPane.showInputDialog("BIENVENIDO A LA SERIE B\n:"
				+"Ingrese hasta que posicion desea llegar en la serie: ");
	
		posdes = Integer.parseInt(Ventana);
		for(pos=1;pos<=posdes;pos++)
		{
			if(i%2==0)
			{
				mensaje=mensaje+i+", ";
				if(k==j)
				{
					i++;
					j++;
					k=1;
				}else {
					k++;
				}
			}else {
				mensaje=mensaje+"-"+i+", ";
				i++;
			}
			
		}
		JOptionPane.showMessageDialog(null,mensaje);
		 
	}
}

File: Tareas/T11SerieB.rb
=begin
Nombre: OSCAR IGNACIO GUAJARDO HERNANDEZ
Matricula:1865021
Hora: V4-6 Martes
=end
mensaje = ""
puts 'BIENVENIDO A LA SERIE B'
puts 'Ingrese hasta que posicion desea llegar en la serie:'
n = gets().to_i 
n=n+1
i=1
for i in (1...n)
  if i.modulo(2)==0
    num=(i/2)+1
    j=1
    for j in (1..num)
      mensaje="#{mensaje} #{i}, "
    end
  else
    mensaje = "#{mensaje} -#{i}, "
  end
end

puts mensaje